Addition of 80405 Game Theory

JCao-CMU opened the pull request:
# Summary of Changes

- Please indicate the classes you are modifying:
  - 80-405 （addition of new class）

# Checklist:

Make sure that your contributions:

- [ √] Are accurate and will not mislead the reader
- [√ ] Do not encourage or cause academic integrity violations
- [√ ] Do not offend professors or unfairly criticize the university

JCao-CMU committed:
Create 80405.md

Added 80405:game theory

JCao-CMU committed:
Update README.md

added 80405.md to the directory

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,18 @@
+| Homeworks |   3     |
+
+
+Game theory is a philosophy department course that served as the introduction to basic game theory. The main topics include Minimax Theory, Nash Equilibrium, Sequential Games, Bayesian Games, and Bargaining Theory. Scenarios include zero-sum games, auctions, votes, and sequantial games. Although the class informally assumes some knowledge about decision theory, it is a introductory class, and all ...
```
"an introductory"

mikinty replied:
Are there math prereqs for understanding the course?

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,18 @@
+Game theory is a philosophy department course that served as the introduction to basic game theory. The main topics include Minimax Theory, Nash Equilibrium, Sequential Games, Bayesian Games, and Bargaining Theory. Scenarios include zero-sum games, auctions, votes, and sequantial games. Although the class informally assumes some knowledge about decision theory, it is a introductory class, and all ...
+
+## Lectures
+The lecture besically covers everything about the course material, and attending lectures would be sufficient to complete all homeworks. Lecture notes are provided beforehand (and enouraged to read before the lecture), but those are not in-depth explanations of the course materals bur rather concepts the lecture is going to use to solve some in-class examplar problems. At the last 60-30 minutes o ...
```
*but not bur

mikinty replied:
30-60 is more colloquial than 60-30

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,18 @@
+The lecture besically covers everything about the course material, and attending lectures would be sufficient to complete all homeworks. Lecture notes are provided beforehand (and enouraged to read before the lecture), but those are not in-depth explanations of the course materals bur rather concepts the lecture is going to use to solve some in-class examplar problems. At the last 60-30 minutes o ...
+
+## Homeworks
+Homeworks are the only element that count as the grade in this course. Yes, this course do not have exams. Homeworks is typically composed of 3-4 problems that covers the topic of interest, and it is often a variant of the problems we solved in the class. There are about 5 homework throughout the semester, and each homework is equally distributed. 
```
Is the grading for homework harsh or easy?

mikinty replied:
"5 homework [assignments]..."

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,18 @@
+## Homeworks
+Homeworks are the only element that count as the grade in this course. Yes, this course do not have exams. Homeworks is typically composed of 3-4 problems that covers the topic of interest, and it is often a variant of the problems we solved in the class. There are about 5 homework throughout the semester, and each homework is equally distributed. 
+
+## Textbooks
```
*especially because it is assigned

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,18 @@
+Homeworks are the only element that count as the grade in this course. Yes, this course do not have exams. Homeworks is typically composed of 3-4 problems that covers the topic of interest, and it is often a variant of the problems we solved in the class. There are about 5 homework throughout the semester, and each homework is equally distributed. 
+
+## Textbooks
+The textbook of this class is Osborne's **An Introduction to Game Theory**, which you can find in many sources. The lecture was build around the content of this book, and reading this book can be very helpful, expecially it is oftentimes assigned reading. If you are really interested at game theory, this book is going to be useful even after the class. 
```
*interested about game theory

mikinty reviewed (changes requested):
Overall is fine. Would like some more details about justification of difficulty, and grammar fixes.

JCao-CMU committed:
Update 80405.md

Grammar fixes and additional data about grading and prerequisite

JCao-CMU commented:
Addressed. Please check the newly edited version.

mikinty commented:
> Addressed. Please check the newly edited version.

"Expecially" is still misspelled in the last paragraph. See the comment I suggested

JCao-CMU committed:
Update 80405.md

More grammar changes

JCao-CMU commented:
Grammar and typo fixed

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,15 @@
+## Homeworks
+Homeworks are the only element that counts as the grade in this course. Yes, this course does not have exams. Homeworks is typically composed of 3-4 problems that cover the topic of interest, and it is often a variant of the problems we solved in the class. There are about 5 homework assignments throughout the semester, and each homework is equally weighted and spaced throughout the semester. Eac ...
+## Textbooks
+The textbook of this class is Osborne's **An Introduction to Game Theory**, which you can find in many sources. The lecture was built around the content of this book, and reading this book can be very helpful, especially it is oftentimes assigned reading. If you are interested in game theory, this book is going to be useful even after the class.
```
```suggestion
The textbook of this class is Osborne's **An Introduction to Game Theory**, which you can find in many sources. The lecture was built around the content of this book, and reading this book can be very helpful, especially because it is oftentimes assigned reading. If you are interested in game theory, this book is going to be useful even after the class.
```

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,15 @@
+
```
```suggestion
```

mikinty replied:
extra space

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,15 @@
+
+# 80-405: Game Theory
+---
```
Why is this here?

mikinty commented on electives/80405.md:
```diff
@@ -0,0 +1,15 @@
+| Exams |   None   |
+
+Game theory is a philosophy department course that served as an introductory to basic game theories. The main topics include Minimax Theory, Nash Equilibrium, Sequential Games, Bayesian Games, and Bargaining Theory. Scenarios include zero-sum games, auctions, votes, prisoner's dilemma, and sequential games. Although the class informally assumes some knowledge about decision theory, it is an intro ...
+## Lectures
```
```suggestion

## Lectures

```

mikinty replied:
Generally we add spaces above and below headers in Markdown

mikinty reviewed (changes requested):
Please see comments

It may be helpful to install a Markdown linter if you can to fix style

JCao-CMU committed:
Update electives/80405.md

Co-authored-by: Michael You <youmichaelc@gmail.com>

JCao-CMU committed:
Update electives/80405.md

Co-authored-by: Michael You <youmichaelc@gmail.com>

JCao-CMU committed:
Update electives/80405.md

Co-authored-by: Michael You <youmichaelc@gmail.com>

mikinty commented:
I'm going to let this pass, but in the future can you please squash commits or make more meaningful changes per commit? Seeing single line changes split into many commits is not helpful, and just pollutes the version history.

You should also be providing more helpful commit messages other than just `Update <file>`

mikinty commented on electives/80405.md:
```diff
@@ -7,7 +7,9 @@
 
 Game theory is a philosophy department course that served as an introductory to basic game theories. The main topics include Minimax Theory, Nash Equilibrium, Sequential Games, Bayesian Games, and Bargaining Theory. Scenarios include zero-sum games, auctions, votes, prisoner's dilemma, and sequential games. Although the class informally assumes some knowledge about decision theory, it is an intro ...
+
 ## Lectures
+
```
If you're going to add the spaces to this header, please do it to the rest of the doc as well

JCao-CMU committed:
Added spacing between headers

Added spacing between headers, removed additional lines in the beginning. Some word choices were changed.
I never used markdown before, and all commits were done in the Github edit function. I have to copy and paste between Github edits and Grammarly...

JCao-CMU commented:
Spacing is added for all headers. 
I never used markdown before, and all commits were done in the Github edit function. I have to copy and paste between Github edits and Grammarly...

mikinty commented:
> Spacing is added for all headers. I never used markdown before, and all commits were done in the Github edit function. I have to copy and paste between Github edits and Grammarly...

I can tell you are using Github edits, but you can still edit the commit title and description to be more than just the default. In addition, you can make more substantial changes each time as well.

For the future I would suggest cloning this locally so that you can make changes accordingly there, with more freedom on version control and linting etc.

I'm actually going to close this PR because there are 8 commits for something that should be probably one commit. Can you please open a new PR and make new commits for the code? I know it's annoying but I don't want 8 commits in the repo if we don't need them.
